### PR TITLE
Add in Slasher Metrics

### LINKED ISSUE #5056: Slasher Prom Metrics
This is an issue to add important prometheus metrics to our slasher, including but not limited to:
- Cache metrics (eviction, size, keys)
- Span cache updating metrics
- Attester slashings detected (surrounding, surrounded, double voting attestations)
- Attestations received per slot (to have a feeling of how much load slasher is enduring vs. how much data the beacon node is aggregating before sending to us)
- Max distance observed (to understand if we have periods of long time between source and target and how other metrics can correlate)

### PR DESCRIPTION
**Pull request by rauljordan**
This resolves #5056.

---

# Description

**Write a summary of the changes you are making**

This PR adds important metrics for us to understand the behavior of our running slasher in production.

In **slasher**
- Num atts received from the beacon node
- Cache evictions
- Lowest and highest observed epochs by slasher
- Latest min/max span distances observed
- Num double proposals and double votes detected
- Num surrounding and surrounded votes detected

In the **beacon node**
- Total slashings received, total slashing successfully inserted in pool
- False positives for slashings received (validator already slashed, exited, etc.)

### CODE REVIEW

File: slasher/detection/metrics.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+		Help: "The # of double vote slashable events detected",
+	})
+	surroundingVotesDetected = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "surrounding_votes_detected",
+		Help: "The # of surrounding slashable events detected",
+	})
+	surroundedVotesDetected = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "surrounded_votes_detected",
+		Help: "The # of surrounded slashable events detected",
+	})
```
Comment (0xKiwi) [inline slasher/detection/metrics.go]: Should we add a "rejected by beacon node pool" metric?

Comment (rauljordan) [reply slasher/detection/metrics.go]: Will add in the beacon node side

File: slasher/detection/attestations/spanner.go [hunk 2]
```diff
@@ -232,6 +245,7 @@ func (s *SpanDetector) updateMaxSpan(ctx context.Context, att *ethpb.IndexedAtte
 	defer traceSpan.End()
 	source := att.Data.Source.Epoch
 	target := att.Data.Target.Epoch
+	latestMaxSpanDistanceObserved.Set(float64(source - target))
```
Comment (0xKiwi) [inline slasher/detection/attestations/spanner.go]:
```suggestion
	latestMaxSpanDistanceObserved.Set(float64(target-source))
```

File: beacon-chain/operations/slashings/metrics.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package slashings
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
```
Comment (terencechain) [inline beacon-chain/operations/slashings/metrics.go]: can we follow prometheus's metric and label naming conventions? (ie. end with `_total` for most of these)

https://prometheus.io/docs/practices/naming/

**Pull request merged**